plugins/Overlay/Overlay.php

```diff
@@ -37,4 +41,39 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = 'General_OverlayRowActionTooltipTitle';
         $translationKeys[] = 'General_OverlayRowActionTooltip';
     }
+
+    /**
+     * Returns if a request belongs to the Overlay page
+     *
+     * Whenever we change the Overlay, or any feature that is available on that page, this list needs to be adjusted
+     * Otherwise it can happen, that the session cookie is sent with samesite=lax, which might break the session in Overlay
+     * See https://github.com/matomo-org/matomo/pull/18648
+     */
+    public static function isOverlayRequest()
+    {
+        $module = Piwik::getModule();
+        $action = Piwik::getAction();
+        $method = Common::getRequestVar('method', '', 'string');
+
+        $isOverlay = $module == 'Overlay';
+        $referrerUrlQuery = parse_url(Url::getReferrer() ?? '', PHP_URL_QUERY);
+        $comingFromOverlay = $referrerUrlQuery && strpos($referrerUrlQuery, 'module=Overlay') !== false;
+        $isPossibleOverlayRequest = (
+            $module === 'Proxy' // JS & CSS requests
+            || ($module === 'API' && 0 === strpos($method, 'Overlay.')) // Overlay API data
+            || ($module === 'CoreHome' && $action === 'getRowEvolutionPopover') // Row evolution
+            || ($module === 'CoreHome' && $action === 'getRowEvolutionGraph') // Row evolution (graph)
+            || ($module === 'CoreHome' && $action === 'saveViewDataTableParameters') // store chart changes (within row evolution & transitions)
+            || $module === 'Annotations' // required to interact with annotations in evolution charts (within row evolution)
+            || ($module === 'Transitions' && $action === 'renderPopover') // Transitions
+            || ($module === 'API' && 0 === strpos($method, 'Transitions.')) // Transitions API data
+            || ($module === 'Live' && $action === 'indexVisitorLog') // Visits Log
```

Comment:
is it actually needed for visits log, transitions and row evolution? Not sure I understand the issue right but would have thought that these requests aren't issued from within an iframe? i'm not too much into the problem though so feel free to ignore it.

Reply:
The cookie is shared between the overlay page and the iframe. If you open the rowevolution the cookie would then be changed to samesite=lax and the iframe would loose access to it. It might not be required at that point but I wanted to avoid possible further regressions for any edge cases.

Reply:
Not sure what to do here. From a security point of view I would maybe rather remove them. I'm thinking for these UIs you could otherwise quite easily force a `None` samesite attribute by using a URL like `/index.php?module=Overlay&action=generalSettings&idSite=1&period=day&date=yesterday&module=CoreAdminHome`.

Not sure how big of an issue it is, maybe it would allow more attacks so would rather restrict it where possible.

Eventually could also check if `module` is twice in the URL etc but these things can get quite complicated and break things for users.

Reply:
@tsteur I've improve the code a bit, so the query of the referrer is parsed. This at least ensures that the module parameter isn't provided twice...